### PR TITLE
Add missing exceptions to JsonSerializer and JsonSerializerOptions

### PR DESCRIPTION
**Pull request by steveharter**
## Summary

Adds missing exception information. Ported from https://github.com/dotnet/runtime/pull/33400/files

### CODE REVIEW

File: xml/System.Text.Json/JsonSerializerOptions.xml [hunk 1]
```diff
@@ -246,6 +246,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
  ]]></format>
         </remarks>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="typeToConvert"/> or its serializable members.</exception>
```
Comment (layomia) [inline xml/System.Text.Json/JsonSerializerOptions.xml]: What about the `InvalidOperationException`: https://github.com/dotnet/runtime/pull/33400/files#diff-bc4374ba33eacef0bf43a48fd7059eb9R117-R119

File: xml/System.Text.Json/JsonSerializer.xml [hunk 2]
```diff
@@ -479,6 +487,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
           ]]></format>
         </remarks>
+        <exception cref="ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
+        <exception cref="ArgumentNullException"><paramref name="inputType" /> is <see langword="null" />.</exception>
```
Comment (layomia) [inline xml/System.Text.Json/JsonSerializer.xml]: We forgot to add this `ArgumentNullException` to the source docs: https://github.com/dotnet/runtime/pull/33400/files#diff-0a82308a189816cb9e1159f4b8af905cR35-R41.

Comment (steveharter) [reply xml/System.Text.Json/JsonSerializer.xml:491]: I added this to a local todo for the next time I touch this file (shouldn't be long)

File: xml/System.Text.Json/JsonSerializerOptions.xml (hunk 1 above)
Comment (steveharter) [reply xml/System.Text.Json/JsonSerializerOptions.xml:250]: Added. Thanks

Comment (carlossanlop) [reply xml/System.Text.Json/JsonSerializerOptions.xml:250]: I wonder why this exception was not ported automatically. @steveharter do you see the exception in the xml generated after you built runtime?

File: xml/System.Text.Json/JsonSerializer.xml [hunk 3]
```diff
@@ -519,8 +530,10 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
           ]]></format>
         </remarks>
+		<exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/></exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:533]: nit - spacing:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/></exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 4]
```diff
@@ -73,6 +73,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the span beyond a single JSON value.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:76]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 5]
```diff
@@ -124,6 +125,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the string beyond a single JSON value.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:128]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 6]
```diff
@@ -182,6 +184,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 A value could not be read from the reader.</exception>
         <exception cref="T:System.ArgumentException">
           <paramref name="reader" /> is using unsupported options.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:187]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 7]
```diff
@@ -230,6 +233,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the span beyond a single JSON value.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:236]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 8]
```diff
@@ -282,6 +286,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the string beyond a single JSON value.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:289]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 9]
```diff
@@ -340,6 +345,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 A value could not be read from the reader.</exception>
         <exception cref="T:System.ArgumentException">
           <paramref name="reader" /> uses unsupported options.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:348]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 10]
```diff
@@ -390,6 +396,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the stream.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:399]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="returnType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 11]
```diff
@@ -440,6 +447,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 -or-
 
 There is remaining data in the stream.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:450]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 12]
```diff
@@ -479,6 +487,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
           ]]></format>
         </remarks>
+        <exception cref="ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:490]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml (hunk 2 above)
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:491]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="inputType" /> is <see langword="null" />.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 13]
```diff
@@ -479,6 +487,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
...
+        <exception cref="ArgumentNullException"><paramref name="inputType" /> is <see langword="null" />.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:492]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 14]
```diff
@@ -519,8 +530,10 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
           ]]></format>
         </remarks>
+		<exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/></exception>
         <exception cref="T:System.ArgumentNullException">
-          <paramref name="writer" /> is <see langword="null" />.</exception>
+          <paramref name="writer" /> or <paramref name="inputType" /> is <see langword="null" />.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:536]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 15]
```diff
@@ -562,6 +575,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
           ]]></format>
         </remarks>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:578]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 16]
```diff
@@ -606,6 +620,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
         </remarks>
         <exception cref="T:System.ArgumentNullException">
           <paramref name="writer" /> is <see langword="null" />.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:623]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 17]
```diff
@@ -647,8 +662,10 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
  ]]></format>
         </remarks>
+        <exception cref="ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:665]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 18]
```diff
@@ -647,8 +662,10 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
...
         <exception cref="T:System.ArgumentNullException">
           <paramref name="utf8Json" /> or <paramref name="inputType" /> is <see langword="null" />.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:668]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 19]
```diff
@@ -694,6 +711,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
         </remarks>
         <exception cref="T:System.ArgumentNullException">
           <paramref name="utf8Json" /> is <see langword="null" />.</exception>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:714]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 20]
```diff
@@ -731,6 +749,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
  ]]></format>
         </remarks>
+        <exception cref="ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:752]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="inputType"/> is not compatible with <paramref name="value"/>.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 21]
```diff
@@ -731,6 +749,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
...
+        <exception cref="ArgumentNullException"><paramref name="inputType" /> is <see langword="null" />.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:753]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="inputType" /> is <see langword="null" />.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 22]
```diff
@@ -731,6 +749,9 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
...
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:754]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="inputType"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 23]
```diff
@@ -770,6 +791,7 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
  ]]></format>
         </remarks>
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializer.xml:794]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <typeparamref name="TValue"/> or its serializable members.</exception>
```

File: xml/System.Text.Json/JsonSerializerOptions.xml [hunk 24]
```diff
@@ -246,6 +246,8 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
 
  ]]></format>
         </remarks>
+        <exception cref="InvalidOperationException">The configured <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="typeToConvert"/> returned an invalid converter.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializerOptions.xml:249]:
```suggestion
        <exception cref="T:System.InvalidOperationException">The configured <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="typeToConvert"/> returned an invalid converter.</exception>
```

File: xml/System.Text.Json/JsonSerializerOptions.xml [hunk 25]
```diff
@@ -246,6 +246,8 @@ For more information, see [How to serialize and deserialize JSON](~/docs/standar
...
+        <exception cref="NotSupportedException">There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/> for <paramref name="typeToConvert"/> or its serializable members.</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonSerializerOptions.xml:250]:
```suggestion
        <exception cref="T:System.NotSupportedException">There is no compatible <see cref="T:System.Text.Json.Serialization.JsonConverter"/> for <paramref name="typeToConvert"/> or its serializable members.</exception>
```

**Comment by carlossanlop** [comment]
@steveharter Did you use the DocsPortingTool to port these exceptions? If yes, then it has a bug, it should be adding the full name prefixed with the type in the crefs.

**Comment by steveharter** [comment]
@carlossanlop how do I run the build so I can verify locally? Thanks

**Comment by steveharter** [comment]
>  Did you use the DocsPortingTool to port these exceptions? If yes, then it has a bug, it should be adding the full name prefixed with the type in the crefs.

No -- there was already doc for the `JsonSerializer` static class so apparently the tool skips. However, I deleted the `JsonSerializer.xml` file and re-ran the tool but it did not generate the doc for that class although it did for other classes that has missing doc.

**Comment by BillWagner** [comment]
@carlossanlop After the latest changes, the build is green. Do you want to approve and merge?

**Pull request merged**